### PR TITLE
Implement MemsetContinueLoop

### LINKED ISSUE #431: MemsetContinueLoop
Reference: https://github.com/starkware-libs/cairo-lang/blob/efa9648f57568aad8f8a13fbf027d2de7c63c2c0/src/starkware/cairo/common/memset.cairo#L22

### PR DESCRIPTION
**Pull request by har777**
Fixes #431

### CODE REVIEW

File: pkg/hintrunner/zero/zerohint_others.go [hunk 1]
```diff
@@ -30,30 +41,36 @@ func newMemcpyContinueCopyingHint(continueCopying hinter.ResOperander) hinter.Hi
...
-func createMemcpyContinueCopyingHinter(resolver hintReferenceResolver) (hinter.Hinter, error) {
-	continueCopying, err := resolver.GetResOperander("continue_copying")
+func createMemContinueHinter(resolver hintReferenceResolver, memset bool) (hinter.Hinter, error) {
+	var continueTarget hinter.ResOperander
+	var err error
+	if memset {
+		continueTarget, err = resolver.GetResOperander("continue_loop")
+	} else {
+		continueTarget, err = resolver.GetResOperander("continue_copying")
+	}
```
Comment (cicr99) [inline pkg/hintrunner/zero/zerohint_others.go:69]: Is it better to pass as a parameter to this function the name of the variable instead of a boolean? You could directly fetch this variable from the resolver and you wouldn't have to do the if statement, wdyt? 🤔

File: pkg/hintrunner/zero/zerohint_others_test.go [hunk 2]
```diff
@@ -38,14 +38,50 @@ func TestZeroHintMemcpy(t *testing.T) {
...
-					return newMemcpyContinueCopyingHint(ctx.operanders["continue_copying"])
+					return newMemContinueHint(ctx.operanders["continue_copying"], false)
 				},
 				check: func(t *testing.T, ctx *hintTestContext) {
 					allVarValueInScopeEquals(map[string]any{"n": feltInt64(4)})(t, ctx)
 					varValueEquals("continue_copying", feltInt64(1))(t, ctx)
 				},
 			},
 		},
+		"MemsetContinueLoop": {
```
Comment (cicr99) [inline pkg/hintrunner/zero/zerohint_others_test.go:49]: I don't think we need to add new tests for this hint. Notice that the only difference in the hint codes is the name of the variable they are using, which is resolved in the createHinter method. Hence, this is not reflected in the unit tests, as you pass directly the resOperander to the newHint method. At the end, you are not testing anything new here.
Only if you came up with a different test, that actually checks a new edge case, then you can leave that

**Review by cicr99 — COMMENTED** [review]
I think these comments also apply to #433. I'll leave a note there so we don't forget to check

File: pkg/hintrunner/zero/zerohint_others.go (hunk 1 above)
Comment (har777) [reply pkg/hintrunner/zero/zerohint_others.go:69]: Also the `newMemContinueHint` function needs to calculate the name of the hint so some sort of `if` condition is still required!

File: pkg/hintrunner/zero/zerohint_others.go [hunk 3]
```diff
@@ -10,10 +10,21 @@ import (
...
+		name = "MemcpyContinueCopying"
+	}
 	return &GenericZeroHinter{
-		Name: "MemcpyContinueCopying",
+		Name: name,
 		Op: func(vm *VM.VirtualMachine, ctx *hinter.HintRunnerContext) error {
+			// MemsetContinueLoop
+			//> n -= 1
+			//> ids.continue_loop = 1 if n > 0 else 0
+
```
Comment (TAdev0) [inline pkg/hintrunner/zero/zerohint_others.go:26]: i think the formatting standard here is to add `//>` also for empty line

Comment (TAdev0) [inline pkg/hintrunner/zero/zerohint_others.go:26]: i think the formatting standard here is to add `//>` also for empty line

**Review by TAdev0 — APPROVED** [review]
lgtm!
left a minor comment

File: pkg/hintrunner/zero/zerohint_others.go (hunk 3 above)
Comment (cicr99) [reply pkg/hintrunner/zero/zerohint_others.go:26]: We do add that if the empty line belongs to the code of the hint, but do not think that's necessary here where you want to create an actual separation.
The reason we use `//>` instead of simple comment lines is because the hint code could be confused in some cases with actual go code from the vm and so it doesn't appear to be just commented code from the codebase

Comment (TAdev0) [reply pkg/hintrunner/zero/zerohint_others.go:26]: Oh yeah you re right my bad :p

**Review by cicr99 — APPROVED** [review]
